File: js/1689.js
const fs = require('fs');

const inputs = fs.readFileSync('example.txt').toString().split('\n');

const N = +inputs[0];
const arr = [];

for (let i = 1; i <= N; i++) {
  const [a, b] = inputs[i].split(' ').map(Number);
  arr.push([a, 1]);
  arr.push([b, -1]);
}

arr.sort((a, b) => {
  if (a[0] === b[0]) return a[1] - b[1];
  return a[0] - b[0];
});

let ans = 0;
let k = 0;
for (let i = 0; i < arr.length; i++) {
  k += arr[i][1];
  if (k > ans) ans = k;
}

console.log(ans);

File: c++/1689.cpp
#include <algorithm>
#include <iostream>
#include <vector>

using namespace std;

vector<pair<int, int>> v;

int main() {
  freopen("example.txt", "r", stdin);
  int n;
  cin >> n;

  while (n--) {
    int x1, x2;
    cin >> x1 >> x2;
    v.push_back({x1, 1});
    v.push_back({x2, -1});
  }

  sort(v.begin(), v.end());
  for (int i = 0; i < v.size(); i++) {
    cout << v[i].first << ' ' << v[i].second << '\n';
  }

  int ans = 0;

  int k = 0;
  for (int i = 0; i < v.size(); i++) {
    k += v[i].second;
    if (k > ans) ans = k;
  }

  cout << ans;

  return 0;
}